Migrate AbsolutePath and RelativePath to URL

Ryu0118 opened the pull request:
I migrated `AbsolutePath` and `RelativePath` to `Foundation.URL`.

ikesyo commented on Sources/ScipioKit/URL+Extensions.swift:
```diff
@@ -1,8 +1,28 @@
+
+    // ref: https://github.com/swiftlang/swift-tools-support-core/blob/f9b401016b70c6b8409e5c97e74d97513d1a8d02/Sources/TSCBasic/Path.swift#L661-L663
+    var parentDirectory: URL {
+        pathComponents.isEmpty ? self : deletingLastPathComponent()
```
`URL(filePath: "/").pathComponents` returns `["/"]` so this should not be correct.

ikesyo commented on Sources/ScipioKit/Producer/PIF/FrameworkBundleAssembler.swift:
```diff
@@ -111,16 +110,13 @@ struct FrameworkBundleAssembler {
-                .appending(component: header.basename)
-                .asURL
+            from: header,
+            to: headerDir.appending(components: subdirectoryComponents)
```
ref:
- https://github.com/giginet/Scipio/pull/227#discussion_r2184473653
- 5cdaecd399c9b4062e336bee5dc6ec98e4d34b45

`.appending(component: header.basename)` was wrongly deleted. I just suggested to remove an unnecessary empty line.

Ryu0118 commented:
Thank you for your review.
I’ve fixed the `parentDirectory` logic and added a test suite to test both `parentDirectory` and `dirname`.
[2282ca4](https://github.com/giginet/Scipio/pull/227/commits/2282ca4e8ae18b8683a46e9309f021dd3bf6650c)

ikesyo reviewed (approved):
LGTM 👍

ikesyo reviewed (changes requested):
Some tests are still failing 👀

ikesyo commented on Sources/ScipioKit/URL+Extensions.swift:
```diff
@@ -1,8 +1,32 @@
+        return String(path.prefix(upTo: idx))
+    }
+
+    // ref: https://github.com/swiftlang/swift-tools-support-core/blob/f9b401016b70c6b8409e5c97e74d97513d1a8d02/Sources/TSCBasic/Path.swift#L661-L663
+    var parentDirectory: URL {
+        path(percentEncoded: false) == "/" ? self : URL(filePath: dirname, directoryHint: .isDirectory)
+    }
```
`AbsolutePath.parentDirectory` does not append the trailing slash so the behavior was slightly changed, but the difference might be safe. 🤔 

How do you think? @Ryu0118 @giginet

Ryu0118 replied:
Since the name parentDirectory suggests a directory path, I considered adding a trailing slash. However, I aligned the behavior with AbsolutePath just to be safe.

[0002568](https://github.com/giginet/Scipio/pull/227/commits/0002568d0397d89fa0b807401c33557c121b7f55)
